crates/cli/src/utils/mod.rs

```diff
@@ -134,18 +118,15 @@ where
 /// it is interpreted as wei.
 pub fn parse_ether_value(value: &str) -> Result<U256> {
     Ok(if value.starts_with("0x") {
-        U256::from_str(value)?
+        U256::from_str_radix(value, 16)?
     } else {
-        U256::from(LenientTokenizer::tokenize_uint(value)?)
+        alloy_dyn_abi::DynSolType::coerce_str(&alloy_dyn_abi::DynSolType::Uint(256), value)?
+            .as_uint()
+            .wrap_err("Could not parse ether value from string")?
+            .0
```

Comment:
this got more complex now, how do we simplify it back?

Reply:
A move to `SolCall` should simplify this—will be done on a followup